### PR TITLE
iox-#1755 Add log bin and log raw

### PR DESCRIPTION
**Pull request by elBoberido**
## Pre-Review Checklist for the PR Author

1. [x] Code follows the coding style of [CONTRIBUTING.md][contributing]
1. [x] Tests follow the [best practice for testing][testing]
1. [x] Changelog updated [in the unreleased section][changelog] including API breaking changes
1. [x] Branch follows the naming format (`iox-123-this-is-a-branch`)
1. [x] Commits messages are according to this [guideline][commit-guidelines]
    - [x] Commit messages have the issue ID (`iox-#123 commit text`)
    - [x] Commit author matches [Eclipse Contributor Agreement][eca] (and ECA is signed)
1. [x] Update the PR title
   - Follow the same conventions as for commit messages
   - Link to the relevant issue
1. [x] Relevant issues are linked
1. [x] Add sensible notes for the reviewer
1. [x] All checks have passed (except `task-list-completed`)
1. [x] All touched (C/C++) source code files from `iceoryx_hoofs` are added to `./clang-tidy-diff-scans.txt`
1. [x] Assign PR to reviewer

[commit-guidelines]: https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
[eca]: http://www.eclipse.org/legal/ECA.php
[contributing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/CONTRIBUTING.md#coding-style
[testing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/concepts/best-practice-for-testing.md
[changelog]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/release-notes/iceoryx-unreleased.md

## Notes for Reviewer

This PR adds logging supports for binary format and raw format. These features were not implemented when the logger was refactored.

## Checklist for the PR Reviewer

- [x] Commits are properly organized and messages are according to the guideline
- [x] Code according to our coding style and naming conventions
- [x] Unit tests have been written for new behavior
- [x] Public API changes are documented via doxygen
- [x] Copyright owner are updated in the changed files
- [x] All touched (C/C++) source code files from `iceoryx_hoofs` have been added to `./clang-tidy-diff-scans.txt`
- [x] PR title describes the changes

## Post-review Checklist for the PR Author

1. [x] All open points are addressed and tracked via issues

## References

- Relates to #1755

### CODE REVIEW

**Comment by elBoberido** [comment]
@FerdinandSpitzschnueffler @mossmaurice @dkroenke @MatthiasKillat @elfenpiff @budrus feel free to add yourself as reviewer

File: iceoryx_platform/win/include/iceoryx_platform/unistd.hpp [hunk 1]
```diff
@@ -22,11 +22,17 @@
 
 #include <io.h>
 #include <process.h>
+#include <stdio.h>
 #include <vector>
 
 #define IOX_SEEK_SET SEEK_SET
 #define _SC_PAGESIZE 1
-#define STDERR_FILENO 2
+#ifndef STDOUT_FILENO
```
Comment (FerdinandSpitzschnueffler) [inline iceoryx_platform/win/include/iceoryx_platform/unistd.hpp:30]: From https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/fileno?view=msvc-170:
"If stdout or stderr is not associated with an output stream (for example, in a Windows application without a console window), the file descriptor returned is -2."
Do we want this behavior?

File: iceoryx_hoofs/reporting/include/iox/log/logstream.hpp [hunk 2]
```diff
@@ -32,15 +32,28 @@ class LogStream;
 
 /// @brief Helper struct to log in hexadecimal format
 template <typename T>
+// NOLINTJUSTIFICATION
```
Comment (FerdinandSpitzschnueffler) [inline iceoryx_hoofs/reporting/include/iox/log/logstream.hpp:35]: Can this be removed?

File: iceoryx_platform/win/include/iceoryx_platform/unistd.hpp (hunk 1 above)
Comment (elBoberido) [reply iceoryx_platform/win/include/iceoryx_platform/unistd.hpp:30]: On the other hand, the man page is not clear regarding of locking of `fileno` calls. Therefore I revert the change and just use the numbers

**Review by FerdinandSpitzschnueffler — APPROVED** [review]

**Review by mossmaurice — APPROVED** [review]
LGTM, thanks!

**Pull request merged**